pkg/segment/structs/evaluationstructs.go

```diff
@@ -363,12 +363,22 @@ func (self *SortExpr) ReleaseProcessedSegmentsLock() {
 func (self *BoolExpr) Evaluate(fieldToValue map[string]utils.CValueEnclosure) (bool, error) {
 	// TODO: Implement the operator in the switch cases below, and replace the if statements with case statements
 	switch self.ValueOp {
-	case "searchmatch":
-		fallthrough
 	case "isnotnull":
-		fallthrough
-	case "isnum":
-		return false, fmt.Errorf("BoolExpr.Evaluate: does support using this operator: %v", self.ValueOp)
+		fields := self.GetFields()
+		if len(fields) == 0 {
+			return false, fmt.Errorf("BoolExpr.Evaluate: No fields found for isnotnull operation")
+		}
+
+		value, exists := fieldToValue[fields[0]]
+		if !exists {
+			return false, fmt.Errorf("BoolExpr.Evaluate: Field '%s' not found in data", fields[0])
+		}
+		if value.Dtype != utils.SS_DT_BACKFILL {
+			return true, nil
+		}
+		return false, nil
+	case "searchmatch", "isnum":
+		return false, fmt.Errorf("BoolExpr.Evaluate: does not support using this operator: %v", self.ValueOp)
 	}
```

Comment:
Add a `default` branch to catch anything we don't expect. And the error on line 381 should clarify that these are valid operators that we don't yet support, while the error in the `default` branch should indicate the operator is invalid

Reply:
Adding a default check here would make the code below unreachable since there are more commands for both `boolOp` and `valueOp` that we support that are not part of this switch statement. I can refactor the method and push it to the same PR so we can also have a default case.

Reply:
You're right, I didn't realize that. Sure, as long as we have some of detecting when an invalid operation is specified it should be good

Reply:
I have cleaned up some code and added a switch case statement to handle all implemented commands, and it looks like we are already handling the default case in https://github.com/siglens/siglens/blob/6884e6a330476401fbcec23e45120636d83dd3af/pkg/segment/structs/evaluationstructs.go#L548
https://github.com/siglens/siglens/blob/6884e6a330476401fbcec23e45120636d83dd3af/pkg/segment/structs/evaluationstructs.go#L558  https://github.com/siglens/siglens/blob/6884e6a330476401fbcec23e45120636d83dd3af/pkg/segment/structs/evaluationstructs.go#L568